### ISSUE TITLE
Accuracy on previous tasks drops dramatically

### ISSUE THREAD

**Issue by eddielyc** [stony-hub/learning_without_memorizing#1]
Hi, thanks for reproducing this paper. I really appreciate it that the code works fine at the very beginning. However, when conducting incremental learning, accuracy on previous tasks nearly drops to zero and it looks like distillation loss fails to mitigate catastrophic forgetting. I clone this repo and only modified the logging part and I wonder whether there is any tricky setting in the code to yield better results. Thanks

**Comment by userDJX** [comment]
@eddielyc  @stony-hub  你好，学霸，我想复现LwM但是在复现的过程中出现了点问题，可以和你一起讨论一下吗？QQ：1109039558，万分感谢

**Comment by userDJX** [comment]
@eddielyc  虽然我没有跑这篇代码，但是我发现一个尴尬的问题，就是grad_cam_loss中的torch.autograd.grad()函数，对于旧模型进行求导，虽然作者加了加了旧模型.requires_grad_(Flase), 但是我这一块出现bug。

**Comment by stony-hub** [comment]
@eddielyc This is my original code and I didn't reproduction the same result as in the paper either. I guess the values of two hyperparameters beta and gamma could be one of the reasons, which are not showed in the paper. Did you find anything new about this? Thank you.

**Comment by stony-hub** [comment]
@userDJX 是什么bug呀
虽然设置了旧模型.requires_grad_(False)，但是在network中手动把feature设为.requires_grad_(True)，这样最后一层仍然是有梯度的

**Comment by userDJX** [comment]
我刚才排查了一下，发现是版本问题，过老的版本不支持requires_grad_，其次就是过老的版本不支持dim=（2，3），我将他们修改了，然后正在跑。

**Comment by userDJX** [comment]
可能存在的问题是，论文中，是固定了全部的参数，可是在你的代码中，我好像看到了只是固定全连接层以前的参数。我认为这块可能会出现问题。

**Comment by stony-hub** [comment]
我看了一下，网络中的参数都是固定的，只有输出的feature和logit的requires_grad为True，所以应该不是这个问题

**Comment by userDJX** [comment]
你跑出来的结果是多少？我将你的代码中的loss加入后，结果太差了[0.82 0.51 0.39 0.33 0.27 ],我不太清楚你的结果是多少？

**Comment by userDJX** [comment]
@stony-hub 能加一下QQ，讨论一下吗？ 1109039558，谢谢~

**Comment by stony-hub** [comment]
其实我这里也差不多2333

**Comment by stony-hub** [comment]
@eddielyc @userDJX I fixed a little bug in the attention distillation part and used a restrict class-incremental setting. I found that it still faces the catastrophic forgetting problem, but it works pretty well in a task-incremental setting. See function test in approach.py for details. I guess this method can hold the result in specific tasks, the same as in LwF.

**Comment by userDJX** [comment]
你能公布一下你的cifar10的实验准确率嘛？都是中国人，哈哈，没必要用英文~

**Comment by userDJX** [comment]
由于我的torch版本的问题，之前许多项目用的都是这个版本，所以导致我不能升级torch。没有办法跑你的实验~

**Comment by stony-hub** [comment]
96.65 86.93 87.83 90.49 90.86
这个是task-incremental的

**Comment by stony-hub** [comment]
class-incremental和你的差不多，之后就掉下来了

**Comment by zwb170010** [comment]
您好，作者，跑了您的代码，发现之前任务灾难性遗忘，是还没有复现原论文的算法嘛

**Comment by stony-hub** [comment]
> 您好，作者，跑了您的代码，发现之前任务灾难性遗忘，是还没有复现原论文的算法嘛

是的

**Comment by XM-WANG** [comment]
> > 您好，作者，跑了您的代码，发现之前任务灾难性遗忘，是还没有复现原论文的算法嘛
> 
> 是的

这不是原论文的算法复现？那这个库的代码是什么。我直接`python main.py`,  跑之后看到前几个task的准确率都掉光了就跑过来看了看issue

**Comment by stony-hub** [comment]
> > > 您好，作者，跑了您的代码，发现之前任务灾难性遗忘，是还没有复现原论文的算法嘛
> > 
> > 
> > 是的
> 
> 这不是原论文的算法复现？那这个库的代码是什么。我直接`python main.py`, 跑之后看到前几个task的准确率都掉光了就跑过来看了看issue

尝试复现但没有成功，只在任务增量场景下work

**Comment by XM-WANG** [comment]
> > > > 您好，作者，跑了您的代码，发现之前任务灾难性遗忘，是还没有复现原论文的算法嘛
> > > 
> > > 
> > > 是的
> > 
> > 
> > 这不是原论文的算法复现？那这个库的代码是什么。我直接`python main.py`, 跑之后看到前几个task的准确率都掉光了就跑过来看了看issue
> 
> 尝试复现但没有成功，只在任务增量场景下work

好的好的，谢谢您！
